src/test/scala/chiselTests/LayerSpec.scala

```diff
@@ -201,4 +201,37 @@ class LayerSpec extends ChiselFlatSpec with Utils with MatchesAndOmits {
       decl("SublayerWithNoOutputDir", List())
     )()
   }
+
+  "Inline layers" should "generated expected FIRRTL" in {
+    object A extends layer.Layer(layer.Convention.Inline) {
+      object B extends layer.Layer(layer.Convention.Inline)
+    }
+
+    class Foo extends RawModule {
+      val a, b, c, d, e = IO(Input(Bool()))
+      val x = IO(Output(Probe(Bool(), A)))
+      val y1 = IO(Output(Probe(Bool(), A.B)))
+      val y2 = IO(Output(Probe(Bool(), A.B)))
+      val y3 = IO(Output(Probe(Bool(), A.B)))
+      define(x, ProbeValue(a))
+      define(y1, ProbeValue(b))
+      layer.block(A) {
+        define(x, ProbeValue(c))
+        define(y2, ProbeValue(d))
+        layer.block(A.B) {
+          define(y3, ProbeValue(e))
+        }
+      }
+    }
+
+    matchesAndOmits(ChiselStage.emitCHIRRTL(new Foo))(
+      """layer A, inline, "A" :""",
+      s"""layer B, inline, "A${sep}B" :""",
+      "define x = probe(a)",
+      "define y1 = probe(b)",
+      "define x = probe(c)",
+      "define y2 = probe(d)",
+      "define y3 = probe(e)"
+    )()
+  }
```

Comment:
This test is overkill for what it is checking. Can this PR instead add an inline layer D and check that this emits as expected?

Reply:
Trimmed this down 👍 .